csharp/src/Ice/Ice1Parser.cs

```diff
@@ -179,7 +179,7 @@ internal static Identity ParseIdentity(string s)
                          string Facet,
                          InvocationMode InvocationMode,
                          Encoding Encoding,
-                         string AdapterId,
+                         string Location0,
```

Comment:
Why not just `Location`?

Reply:
Location0/location0 is to emphasize the string (if not empty) represents segment 0 of the location.
A plain `location` would suggest it's the full location, possibly URI-encoded as in a/b/c.

And alternative could be for this parsing to actually return the location as a IReadOnlyList<string>.